File: Project/ShoppingCart/ShoppingCart/Models/Payment.cs
﻿using ShoppingCart.Enum;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ShoppingCart.Models
{
    public class Payment
    {
        #region Properties
        public int Id { get; set; }

        [ForeignKey("Order")]
        public int OrderId { get; set; }

        [Required]
        [EnumDataType(typeof(PaymentMethod), ErrorMessage = "Payment Method value doesn't exist within enum")]
        public PaymentMethod PayMethod { get; set; }

        [StringLength(250)]
        public string Note { get; set; }

        [Required]
        [DataType(DataType.Currency)]
        public double Amount { get; set; }

        [Required]
        [DataType(DataType.Date)]
        public DateTime Date { get; set; }
        #endregion
    }
}

File: Project/ShoppingCartClient/src/app/Models/Payment.ts
export class Payment{
    orderId: number;
    payMethod: string;
    note: string;
    amount: number;
    date: Date;

    get _payMethod(){
        if(this.payMethod == '0'){
            return 'Pay On Delivery';
        }
        else if(this.payMethod == '1'){
            return 'Shop Credit';
        }
        else if(this.payMethod == '2'){
            return 'Credit Card';
        }
    }

    // set _payMethod(value: string){
    //     if(this.payMethod == 'Pay On Delivery'){
    //         this.payMethod = '0';
    //     }
    //     else if(this.payMethod == 'Shop Credit'){
    //         this.payMethod = '1';
    //     }
    //     else if(this.payMethod == 'Credit Card'){
    //         this.payMethod = '2';
    //     }
    // }
}